### PR TITLE
fix: Improve font rendering

### PR DESCRIPTION
**Pull request by scottc**
Hi

haxcorp 4089 is designed to be used at 12px without anti-aliasing, with integer scaling,
so that the glyph's vector will rasterize and align with the physical pixels for sharp font edges when displayed on digital displays.

But 12px is generally too small for accessibility purposes, so 24 is the minimum size that we can use on 96ppi (pixels per inch) displays.

Here is a comparison screenshot:

Make sure to view the image at full 100% zoom so the browser does not mangle the scaling with bilinear scaling artifacts.

font-rendering

https://user-images.githubusercontent.com/323398/100874802-108e1e00-34e0-11eb-9e41-fb6a8d086e63.png

Screenshot from 2020-12-02 19-44-06

https://user-images.githubusercontent.com/323398/100876581-88f5de80-34e2-11eb-8bda-a648299319ce.png

- Lobster

### CODE REVIEW

File: client/src/gui/views/update.rs [hunk 1]
```diff
@@ -61,7 +61,7 @@ impl UpdateView {
                         Button::new(
                             skip_button_state,
                             Text::new("Skip")
-                                .size(30)
+                                .size(FONT_SIZE_3)
```
Comment (Songtronix) [inline client/src/gui/views/update.rs:64]: you missed to import of `FONT_SIZE_3`.

**Comment by Songtronix** [comment]
everything is looking alright except the news. They look ridiculously big. Sadly the only way to solve that would be to replace the font or keep it as is.

**Comment by scottc** [comment]
> everything is looking alright except the news. They look ridiculously big. Sadly the only way to solve that would be to replace the font or keep it as is.

Agreed, 12/24/36px is pretty restrictive, and doesn't suit well for paragraphs, which you would expect to be about 16~18px, with reasonable line spacing. And the headings to be about 50% larger by comparison (24px?).

A generic sans-serif font with anti-aliasing might be good alternative for paragraphs of text for ease of readability, maybe ask for pfau's opinion?

Anti-aliasing is disabled for technical reasons?

File: client/src/gui/views/update.rs (hunk 1 above)
Comment (scottc) [reply client/src/gui/views/update.rs:64]: Added missing import.

**Comment by Songtronix** [comment]
> Anti-aliasing is disabled for technical reasons?

yes, see https://github.com/hecrj/iced/issues/537
With antialising enabled I wouldn't be able to use airshipper. I will see if we can find a better font.

**Comment by scottc** [comment]
We use OpenSans-Regular for in game chat, and looks like a good sans-serif font, suitable for paragraph text.

**Comment by Songtronix** [comment]
@scottc would you want to replace it?

**Comment by scottc** [comment]
Sure, i'll add the changes to the MR.

**Comment by scottc** [comment]
Updated MR, changes as attached screenshot.

Screenshot from 2020-12-09 18-36-45

Would be nice if the markdown headings were rendered as haxrcorp 4089 in size 24 & 36 for respective level 2 and 3 headings.

**Comment by scottc** [comment]
yikes, the windows build failed again, on that same import.

I'll fix it up.

File: CHANGELOG.md [hunk 2]
```diff
@@ -11,6 +11,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## Fixed
 
+- Improved font rendering
```
Comment (Songtronix) [inline CHANGELOG.md:14]: Could you reference the pull request like in the rest of the changelog?

File: client/src/gui/components/changelog.rs [hunk 3]
```diff
@@ -1,7 +1,6 @@
 use crate::{consts, gui::views::default::DefaultViewMessage, net, Result};
 use iced::{scrollable, Element};
 use serde::{Deserialize, Serialize};
-
```
Comment (Songtronix) [inline client/src/gui/components/changelog.rs:4]: was that intended?

**Review by Songtronix — APPROVED** [review]

File: client/src/gui/components/changelog.rs (hunk 3 above)
Comment (scottc) [reply client/src/gui/components/changelog.rs:4]: explictly by me, no.
implicitly by cargo fmt, yes.

File: CHANGELOG.md (hunk 2 above)
Comment (scottc) [reply CHANGELOG.md]: Added PR ref

**Comment by Songtronix** [comment]
Thanks!

**Pull request merged**